x-pack/plugins/security/server/audit_trail/audit_trail_service.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Subject, combineLatest, of } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import { KibanaRequest, Logger } from 'src/core/server';
+
+import {
+  AuditEvent,
+  AuditEventDecorator,
+  AuditorFactory,
+  LoggingServiceSetup,
+  AuditTrailSetup,
+  HttpServiceSetup,
+  OnPreResponseInfo,
+} from 'src/core/server';
+import { AuditTrailClient } from './audit_trail_client';
+
+import { SecurityPluginSetup } from '../';
+import { SpacesPluginSetup } from '../../../spaces/server';
+import { SecurityLicense } from '../../common/licensing';
+import { ConfigType } from '../config';
+
+interface SetupParams {
+  license: Pick<SecurityLicense, 'features$'>;
+  config: ConfigType['audit'];
+  getCurrentUser: SecurityPluginSetup['authc']['getCurrentUser'];
+  getSpacesService(): Pick<SpacesPluginSetup['spacesService'], 'getSpaceId'> | undefined;
+  logging: Pick<LoggingServiceSetup, 'configure'>;
+  auditTrail: Pick<AuditTrailSetup, 'register'>;
+  http: Pick<HttpServiceSetup, 'registerOnPostAuth' | 'registerOnPreResponse'>;
+}
+
+export class AuditTrailService {
+  private readonly event$ = new Subject<AuditEvent>();
+
+  constructor(private readonly logger: Logger) {}
+
+  public setup({
+    license,
+    config,
+    logging,
+    http,
+    auditTrail,
+    getCurrentUser,
+    getSpacesService,
+  }: SetupParams) {
+    combineLatest([this.event$.asObservable(), license.features$])
+      .pipe(
+        filter(
+          ([event, licenseFeatures]) =>
+            licenseFeatures.allowAuditLogging && filterEvent(event, config.ignore_filters)
+        )
+      )
+      .subscribe(([{ message, ...other }]) => this.logger.info(message, other));
+
+    const depsApi = {
+      getCurrentUser,
+      getSpaceId: (request: KibanaRequest) => getSpacesService?.()?.getSpaceId(request)!,
+    };
+    const auditorFactory: AuditorFactory = {
+      asScoped: (request) => {
+        return new AuditTrailClient(request, this.event$, depsApi);
+      },
+    };
+    auditTrail.register(auditorFactory);
+
+    http.registerOnPostAuth((request, response, t) => {
+      if (request.auth.isAuthenticated) {
+        auditorFactory.asScoped(request).add(httpRequestEvent, { request });
+      }
+      return t.next();
+    });
+    http.registerOnPreResponse((request, preResponseInfo, t) => {
+      if (request.auth.isAuthenticated) {
+        auditorFactory.asScoped(request).add(httpRequestEvent, { request, preResponseInfo });
+      }
+      return t.next();
+    });
+
+    logging.configure(
+      of({
+        appenders: {
+          auditTrailAppender: config.appender ?? {
```

Comment:
That's what we're doing. Have a look at the line below were we set the log level to "off" unless an appender is configured

Reply:
That's what we're doing. Have a look at the line below were we set the log level to "off" unless an appender is configured

Reply:
My understanding is that this will turn off the audit logger if `xpack.security.audit.enabled` is `false`, not whether the appender itself is configured.